### PR TITLE
fraction refactor to fix bug

### PR DESCRIPTION
**Pull request by evykassirer**
this is almost entirely moving code around woo

### CODE REVIEW

File: lib/expression/step-solver/Fraction.js [hunk 1]
```diff
@@ -108,117 +29,23 @@ Fraction.simplifyFraction = function(fraction) {
...
-    else {
-      return new NodeStatus(node);
-    }
-  }
-  else if (NodeType.isConstant(secondArg)) {
-    if (Fraction.isIntegerFraction(firstArg)) {
-      constNode = secondArg;
-      fractionNode = firstArg;
+  // If it's not an integer fraction, all we can do is simplify signs
+  if (!NodeType.isIntegerFraction(fraction)) {
```
Comment (evykassirer) [inline lib/expression/step-solver/Fraction.js]: TODO: add a test for this

File: lib/expression/step-solver/NodeType.js [hunk 2]
```diff
@@ -62,4 +62,13 @@ NodeType.isConstantOrConstantFraction = function(node, allowUnaryMinus=false) {
   }
 }
 
+NodeType.isIntegerFraction = function(node) {
```
Comment (evykassirer) [inline lib/expression/step-solver/NodeType.js]: I'm glad this is finally here

File: lib/expression/step-solver/cancelLikeTerms.js [hunk 3]
```diff
@@ -182,11 +183,17 @@ function cancelTerms(numerator, denominator) {
     return new CancelOutStatus(numerator, denominator, cancelStatus.hasChanged);
   }
   if (NodeType.isUnaryMinus(denominator)) {
-    cancelStatus = cancelTerms(numerator, denominator.args[0]);
+    const cancelStatus = cancelTerms(numerator, denominator.args[0]);
```
Comment (evykassirer) [inline lib/expression/step-solver/cancelLikeTerms.js]: these few lines are the actual bug fix

**Comment by aelnaiem** [comment]
Add tests and then feel free to merge!

**Pull request merged**